### ISSUE TITLE
Pre-submission enquiry

### ISSUE THREAD

**Issue by jstolarek** [openjournals/joss#719]
Our research team is consider writing a JOSS paper about [Links](https://links-lang.org/) and a question came up whether our software is actually within the scope of JOSS. In short, Links is a programming language with several novel/research features already implemented in the language (effect handlers, nested query shredding, session types, a novel approach to first-class polymorphism), and other new features still being developed as part of ongoing research. So basically the language acts as a testbed for new programming language research ideas. We wouldn't call this language production-ready but it is fairly mature (first appeared nearly 15 years ago) but it's not that far from it and we are working on solving real-life problems with Links. The question is whether a research programming language qualifies for a JOSS submission?

**Comment by danielskatz** [comment]
Given that you say "the language acts as a testbed for new programming language research ideas", it seems to me that the answer is yes - this is research software for a part of the CS community, programing language researchers.

On the other hand, I see on the [website](https://links-lang.org), "Links is a functional programming language designed to make web programming easier," which then leads to a question: Is easier web programming part of research? And makes it seem like the answer is no. The [README on the repo](https://github.com/links-lang/links/blob/master/README.md) doesn't help make this case either.

So basically, your statement here and the content of the website & repo don't seem to be aligned, which makes it hard to give one answer.

Another question I wonder about is if there are groups outside the Links team who are using Links for research?  If so, can you point to some papers they have written about their research where they talk about using Links as part of it?  (You don't need to do this here, but would want to do so in the JOSS paper and source code repo's README.)

**Comment by jstolarek** [comment]
> Is easier web programming part of research?

In a sense, yes. When one develops a web application in PHP+MySQL+JavaScript one has to know three different programming languages. With Links you one only has to know and work in one language - Links - and the compiler (well, interpreter) generates SQL queries to execute on the database and JavaScript code to run on the client. So the end result for someone doing web development is that it will be easier (using just one language instead of three) although we are not conducting any rigorous user studies that would somehow measure ease of using Links. Also, web programming with Links is safer due to static type checking (as opposed to runtime errors in SQL and JavaScript).

> Another question I wonder about is if there are groups outside the Links team who are using Links for research?

We are working on a real-world case study where we are developing a web application used for medical research. Part of the features we are developing for Links are driven by their needs, although the development of Links and the said application is happening inside our team. Does that count?

**Comment by danielskatz** [comment]
Given what you've written, I would lean towards saying that Links is not currently research software under the JOSS definition., but I expect other JOSS editors may have different opinions

**Comment by jstolarek** [comment]
> I expect other JOSS editors may have different opinions

Then I presume we should wait for a final yes/no answer based on what other editors say?

**Comment by danielskatz** [comment]
yes - there's also no hurry, since the journal is currently closed for new submissions while we adjust to the current times

**Comment by arfon** [comment]
> With Links you one only has to know and work in one language - Links - and the compiler (well, interpreter) generates SQL queries to execute on the database and JavaScript code to run on the client. So the end result for someone doing web development is that it will be easier (using just one language instead of three) although we are not conducting any rigorous user studies that would somehow measure ease of using Links. Also, web programming with Links is safer due to static type checking (as opposed to runtime errors in SQL and JavaScript).

All of this sounds very useful but doesn't sound much like research to me. I think I agree with @danielskatz here that this sounds like it's likely out of scope for JOSS.

**Comment by jstolarek** [comment]
To clarify as much as possible:

  * using Links to develop web applications is almost certainly not research
  * using Links as a base to develop programming language ideas certainly is research, as shown by a long record of publications in high-profile PL conferences (see [here](https://links-lang.org/)).

**Comment by danielskatz** [comment]
I will mostly repeat what I said before:

Your statement here implies research usage, but the content of the website & repo's README focus on non-research usage, and the research usage seems to entirely be from the team who wrote the software.

In my opinion, to make this something that JOSS would review, the website and README would have to talk about the research aspects of using the software, including kinds of problems that could be researched, and ideally, examples of research publications in those kinds of problems.

**Comment by jstolarek** [comment]
Thanks. That makes sense. I will relay this to my team and see whether we can do better to match JOSS requirements. For now I will close this issue and possibly re-open it in the future if we have further questions after making changes to readme and the webpage.
